src/server.ts

```diff
@@ -553,60 +605,69 @@ function completionHelper(document: TextDocument, textDocumentPosition: Position
 
     const textLine = document.getText().substring(start, end);
 
-    //Check if the string we are looking at is a node
-    if (textLine.indexOf(':') === -1){
-        //We need to add the ":" to load the nodes
-
+    // Check if the string we are looking at is a node
+    if (textLine.indexOf(':') === -1) {
+        // We need to add the ":" to load the nodes
         let newText = '';
 
-        //This is for the empty line case
+        // This is for the empty line case
         const trimmedText = textLine.trim();
-        if (trimmedText.length === 0 || (trimmedText.length === 1 && trimmedText[0] === '-')){
-            //Add a temp node that is in the document but we don't use at all.
+        if (trimmedText.length === 0 || (trimmedText.length === 1 && trimmedText[0] === '-')) {
+            // Add a temp node that is in the document but we don't use at all.
             newText = document.getText().substring(0, start + textLine.length) +
                 (trimmedText[0] === '-' && !textLine.endsWith(' ') ? ' ' : '') + 'holder:\r\n' +
                 document.getText().substr(lineOffset[linePos + 1] || document.getText().length);
 
-        //For when missing semi colon case
-        }else{
-            //Add a semicolon to the end of the current line so we can validate the node
+        // For when missing semi colon case
+        } else {
+            // Add a semicolon to the end of the current line so we can validate the node
             newText = document.getText().substring(0, start + textLine.length) + ':\r\n' + document.getText().substr(lineOffset[linePos + 1] || document.getText().length);
         }
 
         return {
             'newText': newText,
             'newPosition': textDocumentPosition
         };
-
-    }else{
-
-        //All the nodes are loaded
+    } else {
+        // All the nodes are loaded
         position.character = position.character - 1;
+
         return {
             'newText': document.getText(),
             'newPosition': position
         };
     }
-
 }
 
+/**
+ * Like onCompletion, but called only for currently selected completion item
+ * Provides additional information about the item, not just the keyword
+ */
 connection.onCompletionResolve(completionItem => customLanguageService.doResolve(completionItem));
 
+/**
+ * Called when the user hovers with their mouse over a keyword in VS Code
+ * Returns an informational tooltip
+ */
 connection.onHover(textDocumentPositionParams => {
     const document = documents.get(textDocumentPositionParams.textDocument.uri);
 
-    if (!document){
+    if (!document) {
         return Promise.resolve(void 0);
     }
 
     const jsonDocument = parseYAML2(document.getText());
     return customLanguageService.doHover(jsonLanguageService, document, textDocumentPositionParams.position, jsonDocument);
 });
 
+/**
+ * Called when the code outline in VS Code needs to be populated
```

Comment:
I don't think we can explicitly say VS Code here because there are multiple clients available (Eclipse Che, Theia, Atom, etc) that all use this language server